README.md

```diff
@@ -17,240 +20,154 @@ of the [Privacy Community Group](https://privacycg.github.io/).
-the expiry of the logged in session nor the act of logging out.
+### Representation and Storage
 
-Cookies and other kinds of storage may carry login state but there is no
-way to tell general storage and authentication tokens apart. Persistent
-cookies have an expiry function which could serve as an automatic
-inactivity logout mechanism whereas web storage such as IndexedDB
-doesn’t even have an expiry functionality.
+The website's **self-declared** login status is represented as a three-state value **per origin** with the following possible values:
 
-## Straw Man Proposal
+- `unknown`: the browser has never observed a login nor a logout
+- `logged-in`: a user has logged-in to **an** account on the website
+- `logged-out`: the user has logged-out of **all** accounts on the website
 
-Below we present a straw man proposal for how a web API for logged in
-status could look and work. This is a starting point for a conversation,
-not a fully baked proposal.
+ By **default**, every origin has their login status set to `unknown`.
 
-### API
+The login status represents the **client-side** state of the origin in the browser and can be out-of-date with the **server-side** state (e.g. a user can delete their account in another browser instance). Even then, due to cookie expiration, it is an imperfect representation of the client-side state, in that it may be outdated.
 
-Here’s how the API for recording a login could look:
+It should not be the user agent's responsibility to ensure that client-side login status is reflective of server-side or cookie state. However, this proposal wants to make it as easy as possible for developers following recommended patterns to keep these states synchronized.
 
-```
-navigator.recordLogin(
-    username: non-whitespace string of limited length,
-    credentialTokenType: “cookie”,
-    optionalParams { }
-) –> Promise<void>
-```
+### Setting the Login Status
 
-The returned promise would resolve if the status was set and reject if
-not. The API could potentially take an expiry parameter but here we’re
-assuming that a designated cookie manages the expiry of the login.
+There are several mechanisms that a website can use to set the login status:
 
-Here’s how the API for recording a logout could look:
+> TODO: figure out if we can put these behind user activation.
 
-```
-navigator.recordLogout(optionalUsername) –> Promise<void>
-```
+#### JS API
 
-The optional username parameter highlights that we might want to support
-concurrent logins on the same website which would require the site to
-keep track of who to log out and credential tokens to be scoped to user
-names.
+Here’s how the API for recording a login / logout could look:
 
-Here’s how the API for checking the login status could look:
+```javascript
+partial interface Navigator {
+  Promise<void> setLoggedIn(optional LoginStatusOptions options);
+  Promise<void> setLoggedOut(optional LogoutStatusOptions options);
+};
 
+interface LoginStatusOptions {
```

Comment:
this should probably be a dictionary and should have a `boolean idp` member?

I am undecided on whether logout should also have the idp member or not, hmm